include/mbgl/util/string.hpp

```diff
@@ -13,6 +13,19 @@ inline std::string toString(T t) {
     return std::to_string(t);
 }
 
+inline std::string toString(double num) {
+    if (num - (int)num > 0.0 || num + (int)num < 0.0) {
```

Comment:
Nice, I originally went with this (silly) way because `num` didn’t have a reliable type, but now we can make this better.